### PR TITLE
SubstitutionMap cleanups

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Review by DougGregor — APPROVED** [review]
Nifty, thanks!

File: lib/AST/SubstitutionMap.cpp [hunk 1]
```diff
@@ -586,38 +586,55 @@ SubstitutionMap::combineSubstitutionMaps(SubstitutionMap firstSubMap,
...
+             "@objc-protocol");
+      continue;
+    }
+    // All of the conformances should be concrete.
+    if (!conformance.isConcrete()) {
+      llvm::dbgs() << "Concrete substType type:\n";
+      substType->dump(llvm::dbgs());
+      llvm::dbgs() << "SubstitutionMap:\n";
+      dump(llvm::dbgs());
+      llvm::errs() << "\n";
```
Comment (gottesmm) [inline lib/AST/SubstitutionMap.cpp]: This should be llvm::dbgs(), not llvm::errs(), no?

File: lib/AST/SubstitutionMap.cpp [hunk 2]
```diff
@@ -586,38 +586,55 @@ SubstitutionMap::combineSubstitutionMaps(SubstitutionMap firstSubMap,
...
+    // AnyObject or an @objc protocol.
+    if (conformance.isAbstract() &&
+        substType->isExistentialType()) {
+      auto *proto = conformance.getRequirement();
+      if (!proto->isObjC()) {
+        llvm::dbgs() << "Existential type conforms to something:\n";
+        substType->dump();
         llvm::dbgs() << "SubstitutionMap:\n";
         dump(llvm::dbgs());
+        llvm::errs() << "\n";
```
Comment (gottesmm) [inline lib/AST/SubstitutionMap.cpp]: Same here.

**Pull request merged**